crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -2303,19 +2303,70 @@ impl<'src> Parser<'src> {
         statements
     }
 
-    fn parse_parameter(&mut self, function_kind: FunctionKind) -> ast::Parameter {
+    /// Parses a single parameter for the given function kind.
+    ///
+    /// Matches either the `param_no_default_star_annotation` or `param_no_default`
+    /// rule in the [Python grammar] depending on whether star annotation is allowed
+    /// or not.
+    ///
+    /// Use [`Parser::parse_parameter_with_default`] to allow parameter with default
+    /// values.
+    ///
+    /// [Python grammar]: https://docs.python.org/3/reference/grammar.html
+    fn parse_parameter(
+        &mut self,
+        function_kind: FunctionKind,
+        allow_star_annotation: AllowStarAnnotation,
+    ) -> ast::Parameter {
         let start = self.node_start();
```

Comment:
It should be possible because of the `FunctionKind`. We can then call either `parse_function_parameter` or `parse_lambda_parameter` depending on the enum variant.

Reply:
Okay, yeah but if it's used in there than it probably doesn't make that much sense. I think leaving it as is is fine.